cluster-autoscaler/processors/nodegroupconfig/node_group_config_processor_test.go

```diff
@@ -0,0 +1,194 @@
+			NG:     10 * time.Minute,
+		}
+		assert.Equal(t, res, results[w])
+	}
+	testUnreadyTime := func(t *testing.T, p DelegatingNodeGroupConfigProcessor, c *context.AutoscalingContext, ng cloudprovider.NodeGroup, w Want, we error) {
+		res, err := p.GetScaleDownUnreadyTime(c, ng)
+		assert.Equal(t, err, we)
+		results := map[Want]time.Duration{
+			NIL:    time.Duration(0),
+			GLOBAL: 4 * time.Minute,
+			NG:     11 * time.Minute,
+		}
+		assert.Equal(t, res, results[w])
+	}
+	testUtilizationThreshold := func(t *testing.T, p DelegatingNodeGroupConfigProcessor, c *context.AutoscalingContext, ng cloudprovider.NodeGroup, w Want, we error) {
+		res, err := p.GetScaleDownUtilizationThreshold(c, ng)
+		assert.Equal(t, err, we)
+		results := map[Want]float64{
+			NIL:    0.0,
+			GLOBAL: 0.5,
+			NG:     0.75,
+		}
+		assert.Equal(t, res, results[w])
+	}
+	testGpuThreshold := func(t *testing.T, p DelegatingNodeGroupConfigProcessor, c *context.AutoscalingContext, ng cloudprovider.NodeGroup, w Want, we error) {
+		res, err := p.GetScaleDownGpuUtilizationThreshold(c, ng)
+		assert.Equal(t, err, we)
+		results := map[Want]float64{
+			NIL:    0.0,
+			GLOBAL: 0.6,
+			NG:     0.85,
+		}
+		assert.Equal(t, res, results[w])
+	}
+
+	funcs := map[string]struct {
+		testFn     func(*testing.T, DelegatingNodeGroupConfigProcessor, *context.AutoscalingContext, cloudprovider.NodeGroup, Want, error)
+		globalOpts config.NodeGroupAutoscalingOptions
+		ngOpts     *config.NodeGroupAutoscalingOptions
+	}{
+		"ScaleDownUnneededTime": {
+			testFn: testUnneededTime,
+			globalOpts: config.NodeGroupAutoscalingOptions{
+				ScaleDownUnneededTime: 3 * time.Minute,
+			},
+			ngOpts: &config.NodeGroupAutoscalingOptions{
+				ScaleDownUnneededTime: 10 * time.Minute,
+			},
+		},
+		"ScaleDownUnreadyTime": {
+			testFn: testUnreadyTime,
+			globalOpts: config.NodeGroupAutoscalingOptions{
+				ScaleDownUnreadyTime: 4 * time.Minute,
+			},
+			ngOpts: &config.NodeGroupAutoscalingOptions{
+				ScaleDownUnreadyTime: 11 * time.Minute,
+			},
+		},
+		"ScaleDownUtilizationThreshold": {
+			testFn: testUtilizationThreshold,
+			globalOpts: config.NodeGroupAutoscalingOptions{
+				ScaleDownUtilizationThreshold: 0.5,
+			},
+			ngOpts: &config.NodeGroupAutoscalingOptions{
+				ScaleDownUtilizationThreshold: 0.75,
+			},
+		},
+		"ScaleDownGpuUtilizationThreshold": {
+			testFn: testGpuThreshold,
+			globalOpts: config.NodeGroupAutoscalingOptions{
+				ScaleDownGpuUtilizationThreshold: 0.6,
+			},
+			ngOpts: &config.NodeGroupAutoscalingOptions{
+				ScaleDownGpuUtilizationThreshold: 0.85,
+			},
+		},
+		"MultipleOptions": {
+			testFn: func(t *testing.T, p DelegatingNodeGroupConfigProcessor, c *context.AutoscalingContext, ng cloudprovider.NodeGroup, w Want, we error) {
+				testUnneededTime(t, p, c, ng, w, we)
+				testUnreadyTime(t, p, c, ng, w, we)
+				testUtilizationThreshold(t, p, c, ng, w, we)
+				testUnneededTime(t, p, c, ng, w, we)
+				testUnneededTime(t, p, c, ng, w, we)
```

Comment:
These two additional `testUnneededTime` executions seem to be unnecessary.

Reply:
I added them to test that multiple executions of the same function result in the same result and that calling other functions in the meantime doesn't impact it. Would moving it to separate test case make it more readable?

Reply:
I think that would help, right now it seems more like a copy+paste error than a tested case.

Reply:
moved to separate case